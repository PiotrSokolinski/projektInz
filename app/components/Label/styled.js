import styled from 'styled-components'
import { Colors } from 'themes'

export const StyledLabel = styled.label`
  color: ${props => (props.error ? 'red' : Colors.black)};
  font-weight: 500;
  font-size: 12px;
  line-height: 18px;
  letter-spacing: 0.05em;
  text-transform: uppercase;
`
