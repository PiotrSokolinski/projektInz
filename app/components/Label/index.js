/**
 *
 * Label
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { StyledLabel } from './styled'

const Label = ({ error, label }) => <StyledLabel error={error}>{label}</StyledLabel>

Label.propTypes = {
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
}

Label.defaultProps = {
  error: '',
}

export default Label
