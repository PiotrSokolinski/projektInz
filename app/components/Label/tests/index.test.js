/**
 *
 * Tests for Label
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import Label from '../index'

describe('<Label />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(<Label label="Label" />)
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render with error and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Label error="error" label="Label" />)
    expect(firstChild).toMatchSnapshot()
  })

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render without error and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Label error={null} label="Label" />)
    expect(firstChild).toMatchSnapshot()
  })
})
