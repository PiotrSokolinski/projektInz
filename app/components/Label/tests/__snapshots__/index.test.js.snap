// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Should render with error and match the snapshot 1`] = `
<label
  class="styled__StyledLabel-oaisvy-0 gsmrPq"
>
  Label
</label>
`;

exports[`<Label /> Should render without error and match the snapshot 1`] = `
<label
  class="styled__StyledLabel-oaisvy-0 fCsvBu"
>
  Label
</label>
`;
